src/main/java/transforms/GetPrefixDigitsTransform.java

```diff
@@ -31,39 +31,37 @@
 import org.apache.commons.lang3.StringUtils;
 
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
-@JsonIgnoreProperties(value = {TRANSFORM} )
-public class ConditionalColumnTransform extends Transformation<Map<String, String>> {
-    private final List<String> columns;
+public class GetPrefixDigitsTransform extends Transformation<String> {
+    private final String separator;
 
     /**
-     * Represents a transformation to select columns based on non-empty cells.
-     * Function goes over columns until a non-zero input is found.
-     * @param columns: list of columns to go over in sequential order
+     * Represents a transformation to get the digits at the start of a column (if starts with digits).
+     *
+     * @param separator: separation between digits and
      */
-     @JsonCreator
-     public ConditionalColumnTransform(
-            @JsonProperty( Constants.COLUMNS ) List<String> columns) {
-        this.columns = columns;
+    @JsonCreator
+    public GetPrefixDigitsTransform(
+            @JsonProperty( Constants.SEP ) String separator ) {
+        this.separator = separator;
     }
 
-    @JsonProperty( Constants.COLUMNS )
-    public List<String> getColumns() {
-        return columns;
+    @JsonProperty( Constants.SEP )
+    public String getSeparator() {
+        return separator;
     }
 
-    @Override public Object apply( Map<String, String> row ) {
-        for ( String s : columns ) {
-            String thiscol = row.get(s);
-            if ( StringUtils.isNotBlank(thiscol)) {
-                return thiscol;
-            }
+    @Override
+    public Object apply( String o ) {
+        if ( StringUtils.isBlank( o ) ) {
+            return null;
         }
-        return "";
+        if ( Character.isDigit( o.charAt( 0 ) ) ) {
+            String[] strBadge = o.split( separator );
+            return strBadge[ 0 ].trim();
```

Comment:
You might run into issues here if there's digits plus non-digits before the separator, which would cause the write to fail if this is for an int property type. You might want to do `Integer.parseInt` to ensure you're passing only digits.

Reply:
(or better, `Parsers.parseInt`, which already handles all the exception logic)